lib/usdUfe/ufe/StagesSubject.cpp

```diff
@@ -528,6 +528,24 @@ void StagesSubject::stageChanged(
             Ufe::SceneItem::Ptr sceneItem = Ufe::Hierarchy::createItem(ufePath);
             if (!sceneItem || InAddOrDeleteOperation::inAddOrDeleteOperation()) {
                 sendObjectDestroyed(ufePath);
+
+                // If we are not in an add or delete operation, and a prim is
+                // removed, we need to trigger a subtree invalidation.  This is
+                // necessary in order to prevent stale items from being kept in
+                // the global selection set.
```

Comment:
I did a little testing on this trying to just remove the item from the selection list (since subtree invalidate can be quite destructive). That doesn't work since the prim is invalid (at this time) which means we cannot create a scene item for it. Can we add this to the comment here, something like:

```suggestion
                // If we are not in an add or delete operation, and a prim is
                // removed, we need to trigger a subtree invalidation.  This is
                // necessary in order to prevent stale items from being kept in
                // the global selection set.
                // Note: at the point of this notif the prim is not valid anymore
                // and thus we cannot create a scene item to simply remove
                // the item from selection list.
```